hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/client/impl/LeaseRenewer.java

```diff
@@ -96,7 +96,9 @@ public static LeaseRenewer getInstance(final String authority,
    * @param renewer Instance to be cleared from Factory
    */
   public static void remove(LeaseRenewer renewer) {
-    Factory.INSTANCE.remove(renewer);
+    synchronized (renewer) {
+      Factory.INSTANCE.remove(renewer);
```

Comment:
INSTANCE is `static final` so we should synchronize on `LeaseRenewer.class`. But anyways, isn't remove method already synchronized?

Reply:
I see, synchronizing on the object makes sense.